domain/schema/controller.go

```diff
@@ -482,3 +485,65 @@ BEGIN
     VALUES (4, 5, OLD.key, DATETIME('now'));
 END;`)
 }
+
+func upgradeInfoSchema() database.Delta {
+	return database.MakeDelta(`
+CREATE TABLE upgrade_info (
+    uuid             TEXT PRIMARY KEY,
+    previous_version TEXT NOT NULL,
+    target_version   TEXT NOT NULL,
+    init_time        TIMESTAMP NOT NULL,
+    start_time       TIMESTAMP,
```

Comment:
What's the difference between init and start time?

Reply:
init_time is the time that this row is created, however, the upgrade itself can only start once all controllers have been 'registered'. So a non-null `init_time` with null `start_time` is the equivalent of an upgrade in pending status previously

See here:
https://github.com/juju/juju/blob/3.3/worker/upgradesteps/worker.go#L298